### PR TITLE
Warn user during open airlock repair attempts

### LINKED ISSUE #13724: Repairing open airlocks
**Issue Description**:

An airlock that is opened can is said to be getting repaired, but nothing will happen.

**What did you expect to happen**: 

Either to be told that you can't repair an open airlock, or to be able to finish the repairs

**What happened instead**:

You get the message that you start to repair the airlock, but the timer instantly stops.

**Why is this bad/What are the consequences**:

If a door gets broken while open, it'll be stuck open as the bolts will be dropped, and the broken state stops it from having power, making you unable to fix that without something to damage the door to replace it.

**Steps to reproduce the problem**:

Take any airlock and a weapon with enough force. Bolt open the airlock, and damage it past the integrity_failure threshold.

### PR DESCRIPTION
**Pull request by Stealthii**
## What Does This PR Do
This commit adds a check for non-closed airlocks during the repair logic, and returns the user a warning message that airlocks must be closed for repairs to work. This avoids the animation cycle of an interrupted repair attempt, and clarifies that repairs cannot be attempted in this state.

## Why It's Good For The Game
Fixes #13724, clarifying to a user why an airlock repair is not possible in this state.

## Images of changes
image

## Testing
* Damage an airlock to the point it needs repairs
* Attempt repairs whilst airlock is **not** closed, and observe no animation cycle and a warning message
* Attempt repairs whilst airlock is closed, and repairs work as normal

## Changelog
:cl:
fix: Non-closed airlocks no longer animate a failed repair cycle, and warn the user of their inability to be repaired.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -1149,6 +1149,9 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 					"<span class='notice'>You [welded ? "weld the airlock shut":"unweld the airlock"].</span>")
 				update_icon()
 		else if(obj_integrity < max_integrity)
+			if(icon_state != "closed")
```
Comment (Henri215) [inline code/game/machinery/doors/airlock.dm:1152]:
```suggestion
			if(AIRLOCK_OPEN)
```
I think this is how the airlock.dm file is checking for the airlock state.

Comment (Stealthii) [reply code/game/machinery/doors/airlock.dm:1152]: `AIRLOCK_OPEN` is a define for `3`, and used in `update_icon()` as the state parameter (or calculates) to set `icon_state` before passing to the overlay handler.

The actual state here is stored in `icon_state`, which is what we must check to determine current state. We also want to handle all cases where the door is **not** closed, rather than just open.

Comment (Stealthii) [reply code/game/machinery/doors/airlock.dm:1152]: That suggestion would always return the warning, as `3` is always true 😄

Comment (JimKil3) [reply code/game/machinery/doors/airlock.dm:1152]: Could you check the airlock's current state vs `AIRLOCK_OPEN`? I think that's what Henri meant

Comment (Stealthii) [reply code/game/machinery/doors/airlock.dm:1152]: AIRLOCK_OPEN, along with all the other state defines, are only used in `set_airlock_overlays()`, `do_animate()`, and `open()`, with the purpose of updating the icon via `update_icon()` as identifiers for what icon should be used during animation and post-animation. The actual state exists in `icon_state` as a string, and can only be one of the following:
* "open" (when opened)
* "closed" (when closed)
* "nonexistenticonstate" (when in the process of being opened, closed, emagged, or denying access)

Without an overhaul to airlock state code, the only place where state can be determined is two variables:
* `operating` (none, or one of the DOOR_X num defines if being operated)
* `icon_state` (as above)

**Short version:** Comparing to `AIRLOCK_OPEN` doesn't make sense, as that's used solely within update_icon calls.

Comment (Stealthii) [reply code/game/machinery/doors/airlock.dm:1152]: > Weird. Checking the icon state seems a bit scuffed, but if it's the easiest way to do it then it's probably fine

I agree. An alternative option is to compare layer, or we could look at the `density` boolean, as doors aren't interactable during transition anyway. This would be more compatible with sub-objects of airlock that do fancier work around animation.

**Comment by Stealthii** [comment]
After discussion with @JimKil3 using `icon_state` is problematic, especially for sub-objects that may use alternate names for icon_state.

As airlocks aren't interactable during animation, using `density` from base door here should only attempt repairs when the door is solid.

**Review by SteelSlayer — APPROVED** [review]
